src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectRevisionController.java

```diff
@@ -48,76 +57,94 @@
-    @ApiOperation(value = "Lists a catalog object revisions")
-    @ApiImplicitParams({ @ApiImplicitParam(name = "page", dataType = "integer", paramType = "query", value = "Results page you want to retrieve (0..N)"),
-                         @ApiImplicitParam(name = "size", dataType = "integer", paramType = "query", value = "Number of records per page."),
-                         @ApiImplicitParam(name = "sort", allowMultiple = true, dataType = "string", paramType = "query", value = "Sorting criteria in the format: property(,asc|desc). " +
-                                                                                                                                  "Default sort order is ascending. " + "Multiple sort criteria are supported.") })
-    @ApiResponses(value = @ApiResponse(code = 404, message = "Bucket or catalog object not found"))
-    @RequestMapping(value = "/buckets/{bucketId}/resources/{objectId}/revisions", method = GET)
-    public CatalogObjectMetadataList list(@PathVariable Long bucketId, @PathVariable Long objectId) {
-        return catalogObjectRevisionService.listCatalogObjectRevisions(bucketId, objectId);
+        String decodedName = URLDecoder.decode(name.getName(), "UTF-8");
+        CatalogRawObject objectRevisionRaw = catalogObjectService.getCatalogObjectRevisionRaw(bucketId,
+                                                                                              decodedName,
+                                                                                              commitTime);
+
+        byte[] bytes = objectRevisionRaw.getRawObject();
+
+        ResponseEntity.BodyBuilder responseBodyBuilder = ResponseEntity.ok().contentLength(bytes.length);
+
+        try {
+            MediaType mediaType = MediaType.valueOf(objectRevisionRaw.getContentType());
+            responseBodyBuilder = responseBodyBuilder.contentType(mediaType);
+        } catch (org.springframework.http.InvalidMediaTypeException mimeEx) {
+            log.warn("The wrong content type for object: " + decodedName + ", revisionId:" + commitTime +
+                     ", the contentType: " + objectRevisionRaw.getContentType(), mimeEx);
+            mimeEx.printStackTrace();
+        }
+        return responseBodyBuilder.body(new InputStreamResource(new ByteArrayInputStream(bytes)));
     }
 
-    @ApiOperation(value = "Delete a catalog object's revision", notes = "If the revisionId references the latest revision, it is deleted and the catalog object then points to the previous revision. If the revisionId doesn't references the latest revision, it is simply deleted without any impact on the current catalog object. Returns the deleted CatalogObjectRevision metadata.")
+    @ApiOperation(value = "Lists a catalog object revisions")
     @ApiResponses(value = @ApiResponse(code = 404, message = "Bucket or catalog object not found"))
-    @RequestMapping(value = "/buckets/{bucketId}/resources/{objectId}/revisions/{revisionId}", method = DELETE)
```

Comment:
as Brian said, we will keep the revisions, which could be reused as the default revision. we can confirm that with Brian